### PR TITLE
r11s-driver: Add IRouterliciousDriverPolicies with prefetch toggle

### PR DESCRIPTION
**Pull request by znewton**
Allow host to toggle prefetch in the routerlicious driver depending on their needs and introduce `IRouterliciousDriverPolicies` config param as mentioned in #6195

### CODE REVIEW

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts [hunk 1]
```diff
@@ -22,15 +22,34 @@ import { DocumentService } from "./documentService";
 import { ITokenProvider } from "./tokens";
 import { RouterliciousOrdererRestWrapper } from "./restWrapper";
 
+export interface IRouterliciousDriverPolicies {
```
Comment (jatgarg) [inline packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:25]: I think we should add it in a separate file, maybe like policies.ts or utils maybe.

**Review by jatgarg — APPROVED** [review]

**Review by tanviraumi — APPROVED** [review]

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts [hunk 2]
```diff
@@ -19,18 +19,30 @@ import {
...
+    private readonly driverPolicies: IRouterliciousDriverPolicies;
+
     constructor(
         private readonly tokenProvider: ITokenProvider,
+        driverPolicies: Partial<IRouterliciousDriverPolicies> = {},
     ) {
+        this.driverPolicies = {
+            ...defaultRouterliciousDriverPolicies,
+            ...driverPolicies,
+        };
```
Comment (vladsud) [inline packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:45]: Given that the only property in there is required (not optional), do we need defaults?
Note that in ODSP we make every option optional, such that additions of new policy is always non-breaking change and that driver has some reasonable defaults.

**Comment by vladsud** [comment]
should we move instantiation of prefetch layer into r11 driver? I.e. move implementation into driver-utils package and remove this policy altogether for loader to have less logic?

**Review by vladsud — APPROVED** [review]

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts (hunk 2 above)
Comment (znewton) [reply packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:45]: The parameter into DocumentServiceFactory is `Partial<...>` which makes all the properties optional, thus requiring defaults. I think I read somewhere that it's better to use Partial<> than manually making each interface property optional, but I have no idea where or if it was a reputable source. I'm open to making each property optional within the interface if that's a better practice. Let me know either way :)

**Comment by znewton** [comment]
> should we move instantiation of prefetch layer into r11 driver? I.e. move implementation into driver-utils package and remove this policy altogether for loader to have less logic?

This policy object will soon include more flags, such as `enableBlobAggregation` and `enableSinglePayloadSummaryUpload`, so I would rather not remove this. Also, I think we still want to give hosts the option to disable prefetch if they are concerned about memory consumption more than initial load time.

I might also be misinterpreting which policy you're talking about. If you're talking about IDocumentStorageServicePolicies.caching, I am certainly open to moving the prefetch instantiation layer to within the driver and removing `caching` from that policy object.

**Comment by vladsud** [comment]
I meant moving PrefetchDocumentStorageService out of loader layer into driver-utils layer and directly using it in r11s when needed.

**Comment by znewton** [comment]
@vladsud love it. I'll open an issue and get to it at some point 😄

**Pull request merged**